cli/src/pcluster/models/cluster_resources.py

```diff
@@ -74,6 +75,21 @@ def batch_compute_environment(self):
         """Return Batch compute environment."""
         return self._get_output("BatchComputeEnvironmentArn")
 
+    def get_failure_reason(self):
+        """Reason of the failure when the cluster_status is in CREATE_FAILED."""
+
+        def _is_failed_wait(event):
+            if (
+                event.get("ResourceType") == "AWS::CloudFormation::WaitCondition"
+                and event.get("ResourceStatus") == "CREATE_FAILED"
+            ):
+                return True
+            return False
+
+        stack_events = list(itertools.chain.from_iterable(get_all_stack_events(self.name)))
```

Comment:
Question (non-blocking): Does this need to be realized into a list? I'm not sure, but it seems like you could pass the result of `from_iterable` directly to `filter`?